tests/integration/CatalogClient.test.ts

```diff
@@ -0,0 +1,193 @@
+            parameters: {
+              additionalProp1: "string",
+              additionalProp2: "string",
+              additionalProp3: "string"
+            }
+          }
+        ])
+      )
+    );
+
+    const mockedVersions = {
+      versions: [
+        {
+          dependencies: [
+            {
+              direct: false,
+              hrn: "hrn:here:data:::my-catalog",
+              version: 23
+            }
+          ],
+          partitionCounts: {
+            additionalProp1: 0,
+            additionalProp2: 0,
+            additionalProp3: 0
+          },
+          timestamp: "1516397474657",
+          version: 1
+        },
+        {
+          dependencies: [
+            {
+              direct: false,
+              hrn: "hrn:here:data:::my-catalog",
+              version: 23
+            }
+          ],
+          partitionCounts: {
+            additionalProp1: 0,
+            additionalProp2: 0,
+            additionalProp3: 0
+          },
+          timestamp: "1516397474657",
+          version: 2
+        },
+        {
+          dependencies: [
+            {
+              direct: false,
+              hrn: "hrn:here:data:::my-catalog",
+              version: 23
+            }
+          ],
+          partitionCounts: {
+            additionalProp1: 0,
+            additionalProp2: 0,
+            additionalProp3: 0
+          },
+          timestamp: "1516397474657",
+          version: 3
+        },
+        {
+          dependencies: [
+            {
+              direct: false,
+              hrn: "hrn:here:data:::my-catalog",
+              version: 23
+            }
+          ],
+          partitionCounts: {
+            additionalProp1: 0,
+            additionalProp2: 0,
+            additionalProp3: 0
+          },
+          timestamp: "1516397474657",
+          version: 50
+        }
+      ]
+    };
+
+    // Set the response from Metadata service with the versions info from the catalog.
+    mockedResponses.set(
+      `https://metadata.data.api.platform.here.com/metadata/v1/versions?startVersion=1&endVersion=50`,
+      new Response(JSON.stringify(mockedVersions))
+    );
+
+    // Setup the fetch to use mocked responses.
+    fetchMock.withMockedResponses(mockedResponses);
+
+    // Setup Catalog Client with new OlpClientSettings.
+    const settings = new OlpClientSettings({
+      environment: "here",
+      getToken: () => Promise.resolve("test-token-string")
+    });
+
+    const catalogClient = new CatalogClient(
+      HRN.fromString("hrn:here:data:::test-hrn"),
+      settings
+    );
+
+    const request = new CatalogVersionRequest()
+      .withStartVersion(1)
+      .withEndVersion(50);
+
+    let versionsResponse = await catalogClient
+      .getVersions(request)
+      .catch(error => {
+        console.log(`Error getting catalog versions: ${error}`);
+      });
+
+    assert.isDefined(versionsResponse);
```

Comment:
cuold you please check not only isDefined.? because it can be defined, but "`Error getting catalog versions: ${error}`"

Reply:
fixed